src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.soy

```diff
@@ -0,0 +1,109 @@
+                                           numberOfYears: int]
+                        ]}
+  <section class="section--center mdl-grid mdl-grid--no-spacing mdl-shadow--2dp">
+    <div class="mdl-card__supporting-text mdl-grid mdl-grid--no-spacing">
+      <div class="page-content">
+        {call .scholarshipIntroduction}
+          {param generalInfo: $scholarship.generalInfo /}
+        {/call}
+        {call .scholarshipRequirements}
+          {param requirements: $scholarship.requirements /}
+        {/call}
+        {call .scholarshipApplication}
+          {param applicationNotes: $scholarship.applicationNotes /}
+        {/call}
+      </div>
+    </div>
+  </section>
+{/template}
+
+{template .scholarshipIntroduction}
+  {@param generalInfo: [scholarshipName: string,  
+                         schoolsList: string,
+                         introduction: string,
+                         URL: uri]}
+  <div class="mdl-card mdl-cell mdl-cell--12-col">
+    <div id="scholarship-header">
+      <h4>{$generalInfo.scholarshipName}</h4>
+      <a href={$generalInfo.URL}>Visit official website</a>
+      <p> {$generalInfo.schoolsList} </p>
+    </div>
+
+  </div>
+  <div id="introduction">
+    <p>Introduction: {$generalInfo.introduction}</p>
+  </div>
+{/template}
+
+{template .scholarshipRequirements}
+  {@param requirements: [academicRequirements: string, 
+                         financialRequirements: string,
+                         ethnicityRaceRequirements: string,
+                         genderRequirements: string,
+                         locationRequirements: string,
+                         nationalOriginRequirements: string,
+                         otherRequirements: string]}
+  <div class="mdl-card mdl-cell mdl-cell--10-col">
+    <h4 class="mdl-cell mdl-cell--12-col">Requirements</h4>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> Academic Requirements </h6>
+      <p> {$requirements.academicRequirements} </p>
+    </div>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> Ethnicity Race Requirements </h6> 
+      <p> {$requirements.ethnicityRaceRequirements}</p>
+    </div>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> Financial Race Requirements </h6> 
+      <p> {$requirements.financialRequirements}</p>
+    </div>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> Location Requirements </h6>
+      <p> {$requirements.locationRequirements}</p>
+    </div>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> National Origin Requirements </h6>
+      <p> {$requirements.nationalOriginRequirements}</p>
+    </div>
+    <div class="requirement section__text mdl-cell mdl-cell--10-col-desktop">
+      <h6> Other Requirements </h6>
+      <p>{$requirements.otherRequirements}</p>
```

Comment:
I wanted to simplify this part bu iterating through the keys of my input , but it seems that soy doesn't support $object.$key or $object[$key]? Also I'm not sure if I use a loop how I'm gonna add all headers. Is there any suggestions?

Reply:
So for the object key thing, you can use [soy maps](https://github.com/google/closure-templates/blob/master/documentation/reference/types.md). For each individual headers, you can use a for loop and a call to a template for that header, see [here](https://github.com/google/closure-templates/blob/master/documentation/reference/calls.md);

Reply:
So for the object.key thing, you can use soy [maps](https://github.com/google/closure-templates/blob/master/documentation/reference/types.md). And you can use a soy for loop and template call, like [here](https://github.com/google/closure-templates/blob/master/documentation/reference/calls.md) to render all the headers.